apiserver/pkg/storage/calico/resource.go

```diff
@@ -428,15 +427,16 @@ func (rs *resourceStore) GuaranteedUpdate(
 		if err != nil {
 			return err
 		}
-		revInt, _ := strconv.Atoi(accessor.GetResourceVersion())
+		versionStr := accessor.GetResourceVersion()
+		var revInt uint64
+		if versionStr != "" {
+			revInt, err = strconv.ParseUint(versionStr, 10, 64)
+			return fmt.Errorf("failed to parse non-empty resource version %q key=%q: %w", versionStr, key, err)
```

Comment:
This error check feels like the most contentious change here.  I'm going by the comment above: "Check for Revision no. If not set or less than the current version" implies that it should either be empty or a number we can parse.  Anything else feels like a problem.